src/ringct/rctSigs.cpp

```diff
@@ -1579,13 +1579,31 @@ namespace rct {
       }
     }
 
-    bool verRctNonSemanticsSimpleCached(const rctSig & rv)
+    // I put docstring below in case this function gets refactored into another file
+    /**
+     * @brief This is cached version of verRctNonSemanticsSimple which accepts an expanded TX
+     *
+     * All non-serialized fields of tx.rct_signatures should match values present in tx. More
+     * specifically, the key images at tx.rct_signatures.CLSAGs[x].I should match tx.vin[x]. Also,
+     * tx.rct_signatures.bulletproofs_plus[x].V should match values tx.rct_signatures_outPk
+     * (multiplied by group inverse).
+     *
+     * @param tx transaction as seen externally which contains signature (with BC data) to verify
+     * @return true if test passes
+     * @return false if test does not pass
+     */
+    bool verRctNonSemanticsSimpleCached(const cryptonote::transaction & tx)
     {
+      const rctSig & rv = tx.rct_signatures;
```

Comment:
It looks like this function expects that `rv` was constructed by a call to `expand_transaction_2()` or equivalent. As a sanity check you could add an assert that `!(rv.message == rct::zero())`. It doesn't guarantee anything, but would make tracking down errors easier if this function is misused.

Reply:
This issue is resolved with the latest change to implementation